components/Location.js

```diff
@@ -54,4 +33,30 @@ const styles = StyleSheet.create({
     fontSize: 35,
   },
 });
+
+class Location extends React.Component {
+  static propTypes = {
+    handleCityInput: PropTypes.func,
+    getLocation: PropTypes.func,
+  };
+
+  render() {
+    return (
+      <View style={styles.container}>
+        <ASBox handleCityInput={this.props.handleCityInput} />
+        <TouchableHighlight
+          onPress={this.props.getLocation}
+          underlayColor="rgba(95, 158, 160, 0.3)"
+          style={styles.iconContainer}>
+          <Image
+            style={styles.icon}
+            resizeMode="contain"
+            source={locationIcon}
+          />
+        </TouchableHighlight>
+      </View>
+    );
+  }
+}
+
```

Comment:
You should use one convention. Use class component for every implementation, use function for every implementation or use class when you're gonna use state and life cycle method and function when you only render markup.